ql/time/schedule.cpp

```diff
@@ -27,34 +27,26 @@ namespace QuantLib {
 
     namespace {
 
-        Date nextTwentieth(const Date& d, DateGeneration::Rule rule) {
+        Date nextTwentieth(const Date& d) {
             Date result = Date(20, d.month(), d.year());
             if (result < d)
                 result += 1*Months;
-            if (rule == DateGeneration::TwentiethIMM ||
-                rule == DateGeneration::OldCDS ||
-                rule == DateGeneration::CDS) {
-                Month m = result.month();
-                if (m % 3 != 0) { // not a main IMM nmonth
-                    Integer skip = 3 - m%3;
-                    result += skip*Months;
-                }
+            Month m = result.month();
```

Comment:
It's possible that this is called when rule is `Twentieth`, and in this case the adjustment to IMM month shouldn't be performed, should it?